opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -391,6 +392,11 @@ class WellModelMatrixAdapter : public Dune::AssembledLinearOperator<M,X,Y>
             // Construct scalar product.
             auto sp = Dune::createScalarProduct<Vector,POrComm>(parallelInformation_arg, category);
 
+            // If overlapLayers_ > 1 we are using Restricted Additive Schwarz and therefore need
+            // the residual at overlap DoFs.
+            if (overlapLayers_ > 1)
+                parallelInformation_arg.copyOwnerToAll(istlb, istlb);
```

Comment:
Please explain why this is only needed for more than one overlap layers. Shouldn't the right hand side always be consistent (all procs have the same values?)